### DISCUSSION TITLE
Support sub-millisecond resolution timestamps

### DISCUSSION THREAD

**Discussion by kcajf** [grafana/clickhouse-datasource#479] [Ideas]
We have microsecond-resolution data (`DateTime64('UTC', 6)`). Other data sources in Grafana are able to display this in full resolution. It seems like the clickhouse plugin truncates this to millisecond resolution. Could this be avoided?

**Comment by mshustov** [comment]
Could you open a pr similar to https://github.com/grafana/clickhouse-datasource/pull/462 + https://github.com/grafana/clickhouse-datasource/pull/476 ?

**Comment by SpencerTorres** [comment]
See https://github.com/grafana/clickhouse-datasource/issues/771#issuecomment-2018140420
